lib/puppet/provider/dsc_base_provider/dsc_base_provider.rb

```diff
@@ -349,6 +355,17 @@ def invoke_set_method(context, name, should)
     # notify_reboot_pending if data['rebootrequired'] == true
   end
 
+  # Checks whether the given value is valid as a date
+  #
+  # @param date [String] the value to be checked
+  # @return [Boolean] whether the value is valid or not
+  def valid_date?(date)
+    Puppet::Pops::Time::Timestamp.parse(date)
+    true
+  rescue ArgumentError
+    false
+  end
```

Comment:
to save ourselves a bit of trouble here, we could use this as a begin/rescue block above and, in the `rescue`, write the error via `context.err` and assign the value as `nil`.

Reply:
That way we can:

1. Assign the output in a single go instead of "double" parsing the input
2. Let the user know the return was invalid _and_ send them the error `Timestamp.parse` gave us
3. Reduce complexity for ourselves

Reply:
Done